param.go

```diff
@@ -34,6 +34,8 @@ import (
 //                param.
 type (
 	param interface {
+		String() string
```

Comment:
This is a really well-known signature - can we just embed `fmt.Stringer` instead?

It'd be weird if a handler interface explicitly had `ServeHTTP(http.ResponseWriter, *http.Request)` in its definition instead of embedding `http.Handler`, and this feels the same way to me.